src/app/modules/item/pages/item-details/item-details.component.ts

```diff
@@ -48,23 +48,22 @@ export class ItemDetailsComponent implements OnDestroy, BeforeUnloadComponent {
     map(state => state.isReady && state.data),
   );
 
-  private taskTabs = new Subject<TaskTab[]>();
-  taskTabs$ = combineLatest([ this.taskTabs, this.itemData$.pipe(readyData()) ]).pipe(
+  private tabs = new Subject<TaskTab[]>();
+  tabs$ = combineLatest([ this.tabs, this.itemData$.pipe(readyData()) ]).pipe(
     map(([ tabs, data ]) => {
       const canShowSolution = data.item.permissions.canView === 'solution' || !!data.currentResult?.validated;
       return canShowSolution
         ? tabs
         : tabs.filter(tab => tab.view !== 'solution');
     }),
     map(tabs => tabs.filter(tab => !appConfig.featureFlags.hideTaskTabs.includes(tab.view))),
-    startWith([]),
+    startWith(this.router.url.endsWith('/progress/history') ? [{ view: 'progress', name: 'Progress' }] : []),
```

Comment:
Is it safe enough to rely on `url.endsWith('/progress/history')` ? (are query strings ignored?)

As you do it twice in this file, you should at least move to a function in this file... or even a function at the "item" level to make sure we can fix correctly a route change or a bug we would find.

Reply:
Fixed
Indeed, query strings are part of `router.url`, I used a simpler `.includes()` function and moved it to a dedicated method